src/lib/viewers/media/DashViewer.js

```diff
@@ -505,6 +506,28 @@ class DashViewer extends VideoBaseViewer {
                 this.handleDownloadError(error, downloadURL);
                 return;
             }
+
+            if (
+                normalizedShakaError.code === shaka.util.Error.Code.BAD_HTTP_STATUS &&
+                normalizedShakaError.data[1] === 401 // unauthorized error may be cuased by token expired
+            ) {
+                if (this.retryTokenCount >= MAX_RETRY_TOKEN) {
+                    error.details.error_message = 'Reach refreshing token limit for unauthorized error.';
+                } else {
+                    this.options
+                        .refreshToken()
+                        .then(newToken => {
+                            this.options.token = newToken;
+                            this.player.retryStreaming();
+                        })
+                        .catch(tokenError => {
+                            this.triggerError(tokenError);
```

Comment:
What is the expected UX when this error is triggered? Should this instead re-use the `error` defined on L490 so it's not logged as a generic viewer load error?